spec/support/controller_helpers.rb

```diff
@@ -11,4 +11,18 @@ def current_user
   def user_signed_in?
     !current_user.nil?
   end
+
+  def mock_user(role = :user, **attrs)
+    spy('user', **FactoryGirl.attributes_for(role), **attrs)
+  end
+
+  def mock_user_sign_in(user = mock_user)
+    allow(request.env['warden']).to receive(:authenticate!).and_return(user)
+    # necessary because otherwise ApplicationController#app_setup_check fails
+    allow(User).to receive(:count).and_return(1)
+    # necessary for permissions to work
+    allow(ApplicationController).to receive(:current_user).and_return(user)
+    allow(Ability).to receive(:new).and_return(Ability.new(user))
+    allow(described_class).to receive(:current_user).and_return(user)
```

Comment:
Does this not work by default even though `described_class` inherits from `ApplicationController`, i.e. stubbing methods doesn't cascade through inheritance?